lib/rules/for-direction.js

```diff
@@ -101,20 +101,36 @@ module.exports = {
             }
             return 0;
         }
+
+        /**
+         * get the counter position in test
+         * @param {ASTNode} node the node to check
+         * @returns {"left" | "right"} the position of the counter respect to BinaryExpression
+         */
+        function getCounterPosition(node) {
+            if (node.update.type === "UpdateExpression" || node.update.type === "AssignmentExpression") {
+                if (node.update.argument && node.update.argument.type === "Identifier") {
+                    return node.test.left.name === node.update.argument.name ? "left" : "right";
+                }
+            }
```

Comment:
This doesn't seem to cover AssignmentExpression:

```js
for(var i = 0; 10 > i; i-=1){} // should be error
```

and looks partially redundant with functions that check whether the names match afterward.

What do you think about updating ForStatement(node) to always check both sides instead:

```js
ForStatement(node) {

    if (node.test && node.test.type === "BinaryExpression" && node.update) {
        for (const counterPosition of ["left", "right"]) {
            if (node.test[counterPosition].type !== "Identifier") {
                continue;
            }

            const counter = node.test[counterPosition].name;
            // ...
        }
    }
}
```

Reply:
Yes much better! Do you think it is necessary to add other tests on top of this? I added only one more 
`{ code: "for(var i = 0; 10 > i; i-=1){}", errors: [incorrectDirection] }` after your suggestion.

Reply:
> Do you think it is necessary to add other tests on top of this? I added only one more
> `{ code: "for(var i = 0; 10 > i; i-=1){}", errors: [incorrectDirection] }` after your suggestion.

One test looks fine to me.